src/main/routes/home.ts

```diff
@@ -3,10 +3,26 @@ import { Application } from 'express';
 import { Redis } from 'ioredis';
 
 export default async function (app: Application): Promise<void> {
-  const redis = new Redis(config.get('pcs.redis-connection-string'));
-  await redis.set('mykey', 'value');
   app.get('/', async (req, res) => {
-    const mykey = await redis.get('mykey');
-    res.render('home', { redisKey: mykey });
+    let logMsg = '';
+    if (!app.locals.redis) {
+      try {
+        app.locals.redis = new Redis(config.get('pcs.redis-connection-string'), {
```

Comment:
May be the issue is we are trying connection on every request, try to move to a new method and create a new one if its not null. Something like 

```
let redisClient;

function getRedisClient() {
  if (!redisClient) {
    redisClient = new Redis(config.get('pcs.redis-connection-string'), {
      maxRetriesPerRequest: 3,
    });
    redisClient.on('error', (err) => {
      console.error(`Redis error: ${err.message}`);
    });
  }
  return redisClient;
}
```

Reply:
the code template will change as this would be an abstraction away from the route the validations but will create a new method and a validation for null checks.